syntax = "proto3";

package NPredUnitPb;

import "matrix.proto";
import "layer-config.proto";
import "sim.proto";

message TMessageType {
	enum EMessageType {
		START_SIM = 0;
		INPUT_DATA = 1;
		SERVER_RESPONSE = 2;
		STAT_REQUEST = 3;
	}

	EMessageType MessageType = 1;
}


message TStartSim {
	TSimConfig SimConfig = 1;
	TInputData InputData = 2;
}

message TStatRequest {
	uint32 SimId = 1;
}

message TInputData {
	uint32 SimId = 1;
	TMatrix Data = 2;
}

message TStats {
	TMatrix Membrane = 1;
	TMatrix Activation = 2;
	TMatrix F = 3;
	TMatrix Fc = 4;
}

message TServerResponse {
	enum EResponseType {
		OK = 0;
		SIM_NOT_RUN = 1;
		BUSY = 2;
		ERROR = 3;
		STATS = 4;
	}

	EResponseType ResponseType = 1;
	string Message = 2;
	TStats Stats = 3;
}


